internal/overlord/servstate/manager_test.go

```diff
@@ -320,6 +328,32 @@ func (s *S) stopTestServicesAlreadyDead(c *C) {
 	s.st.Unlock()
 }
 
+func (s *S) TestGracePeriodIsUsed(c *C) {
```

Comment:
This test takes 10 seconds to run, which more than doubles our test run time. :-) Can we shorten it to a couple hundred milliseconds at most? But we also need to ensure it's not flakey.